sfaira/data/base/dataset.py

```diff
@@ -795,17 +789,17 @@ def streamline_metadata(
             if "gene_id_symbols" not in adata_target_ids.var_keys:
                 self.gene_id_symbols_var_key = None
         else:
-            self.adata.var = pd.concat([var_new, self.adata.var], axis=1, ignore_index=True)
             self.adata.var.index = var_new.index
+            self.adata.var = pd.concat([var_new, self.adata.var], axis=1)
         if clean_obs:
             if self.adata.obsm is not None:
                 del self.adata.obsm
             if self.adata.obsp is not None:
                 del self.adata.obsp
             self.adata.obs = obs_new
         else:
-            self.adata.obs = pd.concat([obs_new, self.adata.obs], axis=1, ignore_index=True)
             self.adata.obs.index = obs_new.index
+            self.adata.obs = pd.concat([obs_new, self.adata.obs], axis=1)
```

Comment:
This is fixed on dev now, I also caught column duplication which otherwise throws errors in writing.

Reply:
yep, just saw! nice